### PR TITLE
Integration tests for state-[delete|get|set] hook tools.

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Integration test for work done in PRs:
- #11166 hook tools for state cache get/delete/set 
- #11158 Implement server-side bits for the State/SetState uniter API endpoints
- #11136 Cache unit state
- #11142 Setup mongo collection for server side charm state

## QA steps

```sh
(cd tests ; ./main.sh hook_tools)
```

### CODE REVIEW

File: tests/suites/hook_tools/state_tools.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    model_name="test-state-delete-get-set"
+    file="${TEST_DIR}/${model_name}.txt"
+
+    ensure "${model_name}" "${file}"
+
+    juju deploy cs:~jameinel/ubuntu-lite-7
+    wait_for "ubuntu-lite" "$(idle_condition "ubuntu-lite")"
+
+    juju run --unit ubuntu-lite/0 state-get | grep -q "{}"
+    juju run --unit ubuntu-lite/0 state-set one=two
```
Comment (achilleasa) [inline tests/suites/hook_tools/state_tools.sh]: Don't you need to quote the juju run commands?
E.g.

```juju run --unit ubuntu-lite/0 'state-set one=two'```

Comment (hmlanigan) [reply tests/suites/hook_tools/state_tools.sh]: added

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
intermittent fail in applicationConstraintsSuite.TestConstraintsOpsForSpaceNameChange

$$merge$$

**Pull request merged**